auth.go

```diff
@@ -1,102 +1,111 @@
+	s.UserAuthClaimNonRevMtpAuxHi = nodeAuxAuth.key
+	s.UserAuthClaimNonRevMtpAuxHv = nodeAuxAuth.value
+	s.UserAuthClaimNonRevMtpNoAux = nodeAuxAuth.noAux
 
-// GetPublicSignalsSchema returns schema to parse public inputs
-func (c AuthCircuit) GetPublicSignalsSchema() PublicSchemaJSON {
-	return AuthenticationPublicSignalsSchema
+	return json.Marshal(s)
 }
 
-// PrepareInputs returns inputs for id state circuit as map
-func (c *AuthCircuit) PrepareInputs(in TypedInputs) (map[string]interface{}, error) {
+// AuthPubSignals auth.circom public signals
+type AuthPubSignals struct {
+	Challenge *big.Int         `json:"challenge"`
+	UserState *merkletree.Hash `json:"userState"`
+	UserID    *core.ID         `json:"userID"`
+}
 
-	authInputs, ok := in.(AuthInputs)
-	if !ok {
-		return nil, errors.New("wrong type of input arguments")
-	}
-	inputs := make(map[string]interface{})
-
-	inputs["userID"] = authInputs.ID.BigInt().String()
-	inputs["userState"] = authInputs.State.StateStr()
-
-	inputs["userAuthClaim"] = bigIntArrayToStringArray(authInputs.AuthClaim.Slots)
-
-	inputs["userAuthClaimMtp"] = bigIntArrayToStringArray(PrepareSiblings(authInputs.AuthClaim.Proof.Siblings, AuthenticationLevels))
-	inputs["userAuthClaimNonRevMtp"] = bigIntArrayToStringArray(PrepareSiblings(authInputs.AuthClaimNonRevocationProof.Siblings, AuthenticationLevels))
-
-	inputs["userClaimsTreeRoot"] = authInputs.State.ClaimsRootStr()
-	inputs["userRevTreeRoot"] = authInputs.State.RevocationRootStr()
-	inputs["userRootsTreeRoot"] = authInputs.State.RootOfRootsRootStr()
-
-	inputs["challenge"] = authInputs.Challenge.String()
-	inputs["challengeSignatureR8x"] = authInputs.Signature.R8.X.String()
-	inputs["challengeSignatureR8y"] = authInputs.Signature.R8.Y.String()
-	inputs["challengeSignatureS"] = authInputs.Signature.S.String()
-
-	if authInputs.AuthClaimNonRevocationProof.NodeAux == nil {
-		inputs["userAuthClaimNonRevMtpAuxHi"] = merkletree.HashZero.BigInt().String()
-		inputs["userAuthClaimNonRevMtpAuxHv"] = merkletree.HashZero.BigInt().String()
-		inputs["userAuthClaimNonRevMtpNoAux"] = new(big.Int).SetInt64(1).String() // (yes it's isOld = 1)
-	} else {
-		inputs["userAuthClaimNonRevMtpNoAux"] = new(big.Int).SetInt64(0).String() // (no it's isOld = 0)
-		if authInputs.AuthClaimNonRevocationProof.NodeAux.HIndex == nil {
-			inputs["userAuthClaimNonRevMtpAuxHi"] = merkletree.HashZero.BigInt().String()
-		} else {
-			inputs["userAuthClaimNonRevMtpAuxHi"] = authInputs.AuthClaimNonRevocationProof.NodeAux.HIndex.BigInt().String()
-		}
-		if authInputs.AuthClaimNonRevocationProof.NodeAux.HValue == nil {
-			inputs["userAuthClaimNonRevMtpAuxHv"] = merkletree.HashZero.BigInt().String()
-		} else {
-			inputs["userAuthClaimNonRevMtpAuxHv"] = authInputs.AuthClaimNonRevocationProof.NodeAux.HValue.BigInt().String()
-		}
+// CircuitOutputUnmarshal unmarshal auth.circom public inputs to AuthPubSignals
+func (a *AuthPubSignals) CircuitOutputUnmarshal(data []byte) error {
```

Comment:
How about changing to PubSignalsUnmarshal? Or even just Unmarshal/Parse - it's a method of <circuitName>PubSignals struct. Same for CircuitInputMarshal method of <circuitName>Inputs struct.

Reply:
+1